samples/buildstrategy/buildkit/buildstrategy_buildkit_cr.yaml

```diff
@@ -19,15 +19,21 @@ spec:
           add:
             - CHOWN
       command:
-        - /bin/sh
+        - /bin/chown
       args:
-        - -c
-        - |
-          set -exuo pipefail
-          whoami
-          chown -R 1000:1000 /tekton/home
+        - -R
+        - 1000:1000
+        - /tekton/home
     - name: build-and-push
       image: moby/buildkit:master-rootless
+      securityContext:
+        allowPrivilegeEscalation: true
```

Comment:
As mentioned internally. Im not a fan of setting `allowPrivilegeEscalation` to true in our strategy. 

Reasons are:
- Simplicity: There is no need for this, if someone have PSP enable, then we will provide that knowledge via the nice documentation extension you just provide.
- Generates Noise: PrivilegeEscaltion + SETGID/SETUID are only needed when PSP is enable. Its hard to assume what does the majority of users have, but enabling this config might generate confusion around "if buildkit is fully rootless or not", one needs then to understand things like , all docker `RUN` cmds do indeed run in a user namespace that maps to an unprivileged UID in the host (rootless). If we can avoid this confusion, it will be better.

Reply:
But, it is reality that BuildKit (or rootlesskit) needs it. So, why not being transparent? Without PSP, the strategy should work with and without the flag. With a PSP, the strategy can only work with the flag (and the PSP allowing it but having defaultAllowPrivilegeEscalation set to false).

So, I would specify it like that. We can add a comment to link to the documentation.

Otherwise we could also remove the seccomp and apparmor annotations as they are also only needed if those are enabled.

Reply:
I mean, better having it, not needing it than needing it, not having it. For example, I had to figure out these settings in a PSP use case. Sure, you could argue that it is in the docs, but it does not do any harm here and now it is a sample you can use in both cases, with or without PSP.

Reply:
I understand your point. But I wanted to also express mine.  I think that as long as we document it, we are transparent. So my proposal does not goes against transparency. 

If `container.seccomp.security.alpha.kubernetes.io/step-build-and-push` is also not needed, then we remove it and we document it in the same way as `allowPrivilegeEscalation`.

I dont see too much value on deciding if it should be call "french fries" or "chips" and this will clearly go in that direction, so why dont we ask @sbose78 or @gabemontero to make the call? 👍🏽

Reply:
@qu1queee Can we finalize our decision on this one?

Reply:
@adambkaplan when you say `I'd prefer our build strategy samples assume a secured environment than not.` , are you saying:

- [1] We should expect this extra security configurations to be done in downstream.
- [2] We should include this extra security settings in our strategies.

which one from the above :) ?

Reply:
@adambkaplan thanks, this is very informative. I recall seeing PSP being deprecated soon, but I was not aware of the pod security standards. This is actually concerning, as I think most clusters would like to use the `Restricted` PSS, meaning BuildKit will not manage to work without the `allowPrivilegeEscalation`.